components/server/src/workspace/workspace-starter.ts

```diff
@@ -1129,12 +1136,13 @@ export class WorkspaceStarter {
         envvars.push(contextEnv);
 
         log.debug("Workspace config", workspace.config);
-        if (!!workspace.config.tasks) {
-            // The task config is interpreted by Theia only, there's little point in transforming it into something
+        const tasks = this.ideService.resolveGitpodTasks(workspace);
```

Comment:
nit: The responsibility to add `workspace.config.tasks` should IMO remain with `workspace-starter`. But anyway, let's move forward with this. :+1:

Reply:
It is not so clear to me. When we have IDE service component, its responsibility will be to assemble workspace specification for IDE, including supervisor image and env vars relevant for this version of supervisor. Since tasks env var belongs to supervisor I would imagine that computation will be taken over by IDE service then. Server does not really know what supervisor expects.

Reply:
Hm, fair point. On the other hand it's not supervisors responsibility to ensure the workspace is configured correctly. Can we make a 100% distinction for env vars? :thinking:  I always thought that it's the "workspace starters" responsibility.
Anyway, we should talk about this more often, and earlier. :+1: